zio-flow/shared/src/main/scala/zio/flow/internal/DurableIndexedStore.scala

```diff
@@ -0,0 +1,120 @@
+package zio.flow.internal
+import org.rocksdb.{ColumnFamilyDescriptor, ColumnFamilyHandle, RocksDBException}
+import zio.rocksdb.service.TransactionDB
+import zio.rocksdb.{Transaction, service}
+import zio.schema.Schema
+import zio.schema.codec.ProtobufCodec
+import zio.stream.ZStream
+import zio.{Chunk, Has, IO, Ref, UIO, ZIO, ZLayer}
+
+import java.io.IOException
+
+final case class DurableIndexedStore(transactionDB: service.TransactionDB, colHandles: Ref[List[ColumnFamilyHandle]])
+    extends IndexedStore {
+
+  def addTopic(topic: String): IO[IOException, ColumnFamilyHandle] =
+    for {
+      //TODO : Only catch "Column Family already exists"
+      colFamHandle <-
+        transactionDB
+          .createColumnFamily(new ColumnFamilyDescriptor(ProtobufCodec.encode(Schema[String])(topic).toArray))
+          .catchSome { case _: RocksDBException =>
+            getColFamilyHandle(topic)
+          }
+          .refineToOrDie[IOException]
+      _ <- transactionDB
+             .put(
+               colFamHandle,
+               ProtobufCodec.encode(Schema[String])("POSITION").toArray,
```

Comment:
These constant arrays could be precalculated

Reply:
Done.